test/cli/test_new.rb

```diff
@@ -0,0 +1,257 @@
+require_relative "test_helper"
+require "aruba"
+
+class CLITest < Minitest::Test
+  include Aruba::Api
+  include Helpers
+
+  @@debugging = false
+  ARGV.each do|a|
+    if a == "-vv"
+      @@debugging = true
+    end
+  end
+
+  def setup
+    setup_aruba
+  end
+
+  def test_config
+    skip "aruba log_level adjustment has no apparent effect"
+
+    label = "test-login"
+    username = "user"
+    password = "pass"
+
+    # NOTE: no apparent effect:
+    # configure log level
+    Aruba.configure do |config|
+      config.log_level = :debug
+    end
+
+    # NOTE: cannot convince to work
+    # create login:
+    run_command "kbsecret new login #{label}"
+    type "#{username}"
+    type "#{password}"
+    close_input
+    stop_all_commands
+
+    # retrieve login:
+    run_command_and_stop "kbsecret login -x #{label}"
+    assert_equal "#{label}:#{username}:#{password}", last_command_stopped.output.chomp
+  end
+
+  def test_types
```

Comment:
I think it would be good to break this up into individual methods, like `test_new_login`, `test_new_environment`, etc. There's a lot of conditional logic in here that could be avoided that way.